modules/apps/dynamic-data-mapping/dynamic-data-mapping-test/src/testIntegration/java/com/liferay/dynamic/data/mapping/service/test/BaseTemplateUpgradeProcessTestCase.java

```diff
@@ -0,0 +1,156 @@
+import com.liferay.portal.kernel.workflow.WorkflowConstants;
+import com.liferay.portal.test.log.LogCapture;
+import com.liferay.portal.test.log.LoggerTestUtil;
+import com.liferay.portal.test.rule.Inject;
+import com.liferay.portal.upgrade.registry.UpgradeStepRegistrator;
+import com.liferay.portal.upgrade.test.util.UpgradeTestUtil;
+
+import org.junit.Before;
+
+/**
+ * @author Albert Gomes Cabral
+ */
+public abstract class BaseTemplateUpgradeProcessTestCase {
+
+	@Before
+	public void setUp() throws Exception {
+		_group = GroupTestUtil.addGroup();
+
+		_serviceContext = ServiceContextTestUtil.getServiceContext(
+			_group.getGroupId(), TestPropsValues.getUserId());
+	}
+
+	protected void addDDMTemplate(String filePath) throws Exception {
+		long classPK = ParamUtil.getLong(_serviceContext, "classPK");
+
+		ddmTemplate = _ddmTemplateService.addTemplate(
+			_group.getGroupId(), _portal.getClassNameId(DDMTemplate.class),
+			classPK, _portal.getClassNameId(JournalArticle.class),
+			HashMapBuilder.put(
+				LocaleUtil.US, "DDMTemplate Name"
+			).build(),
+			HashMapBuilder.put(
+				LocaleUtil.US, "DDMTemplate Description"
+			).build(),
+			DDMTemplateConstants.TEMPLATE_TYPE_DISPLAY, StringPool.BLANK,
+			TemplateConstants.LANG_TYPE_FTL, read(filePath), _serviceContext);
+	}
+
+	protected void addFragmentEntry(String filePath) throws Exception {
+		FragmentCollection fragmentCollection =
+			_fragmentCollectionService.addFragmentCollection(
+				_group.getGroupId(), "Fragment Collection", StringPool.BLANK,
+				_serviceContext);
+
+		fragmentEntry = _fragmentEntryService.addFragmentEntry(
+			_group.getGroupId(), fragmentCollection.getFragmentCollectionId(),
+			null, "FragmentEntry Name", null, read(filePath), null, false, null,
+			null, 0, FragmentConstants.TYPE_COMPONENT, null,
+			WorkflowConstants.STATUS_APPROVED, _serviceContext);
+	}
+
+	protected DDMTemplate getDDMTemplate() throws Exception {
+		return _ddmTemplateService.getTemplate(ddmTemplate.getTemplateId());
+	}
+
+	protected FragmentEntry getFragmentEntry() throws Exception {
+		return _fragmentEntryService.fetchFragmentEntry(
+			fragmentEntry.getFragmentEntryId());
+	}
+
+	protected abstract String getUpgradeStepClassName() throws Exception;
+
+	protected String read(String filePath) throws Exception {
+		Class<?> clazz = getClass();
+
+		return StringUtil.read(
+			clazz.getClassLoader(),
+			"com/liferay/dynamic/data/mapping/dependencies/upgrade" + filePath);
+	}
+
+	protected void runTemplateUpgrade() throws Exception {
+		try (LogCapture logCapture = LoggerTestUtil.configureLog4JLogger(
+				getUpgradeStepClassName(), LoggerTestUtil.OFF)) {
+
+			UpgradeProcess upgradeProcess = UpgradeTestUtil.getUpgradeStep(
+				_upgradeStepRegistrator, getUpgradeStepClassName());
+
+			upgradeProcess.upgrade();
+
+			_multiVMPool.clear();
+		}
+	}
+
+	@Inject
+	protected DDMTemplate ddmTemplate;
+
+	@Inject
+	protected FragmentEntry fragmentEntry;
```

Comment:
@alberttgomes, no need to use them as `protected`, please keep them as `private`

Reply:
No? we is using the ddmTemplate and fragmentEntry in `TemplateBrowserSnifferUpgradeProcessTest` ? like suggest here [link](https://github.com/natocesarrego/liferay-portal/pull/1313/commits/7864853a30b42e904540b4faa3f566c092156102#r1256358878)

Reply:
I see your point, @alberttgomes. My code snippet wasn't 100% precise, sorry about that. As a good practice of OOP (Objects Oriented Programming), we should encapsulate the class attributes and access them through its methods get and set, am I clearer now?